sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/DiskEncryptionSet.java

```diff
@@ -109,17 +110,33 @@ interface WithSystemAssignedManagedServiceIdentity {
         }
 
         /**
-         * The stage of the System Assigned (Local) Managed Service Identity enabled disk encryption set allowing to set
+         * The stage of the System Assigned Managed Service Identity enabled disk encryption set allowing to set
          * access role for the key vault.
          */
         interface WithSystemAssignedIdentityBasedAccessOrCreate extends WithCreate {
             /**
-             * Specifies that disk encryption set's system assigned (local) identity should have the given access
+             * Specifies that disk encryption set's system assigned identity should have the given RBAC based access
              * (described by the role) on the current Azure key vault that's associated with it.
+             * Only works for key vaults that use the 'Azure role-based access control' permission model.
+             * If you prefer Access Policy based access for Azure Key Vault (like the examples from Portal or CLI),
+             * instead of calling this method, you may want to call Vault-related methods after creating the
+             * {@link DiskEncryptionSet} instance.
              * @param builtInRole access role to assigned to the disk encryption set's local identity
              * @return the next stage of the definition
              */
-            WithCreate withSystemAssignedIdentityBasedAccessToCurrentKeyVault(BuiltInRole builtInRole);
+            WithCreate withRBACBasedAccessToCurrentKeyVault(BuiltInRole builtInRole);
+
+            /**
+             * Specifies that disk encryption set's system assigned identity should have the RBAC based access
+             * with default {@link BuiltInRole#KEY_VAULT_CRYPTO_SERVICE_ENCRYPTION_USER} on the current Azure key vault
+             * that's associated with it.
+             * Only works for key vaults that use the 'Azure role-based access control' permission model.
+             * If you prefer Access Policy based access for Azure Key Vault (like the examples from Portal or CLI),
+             * instead of calling this method, you may want to call Vault-related methods after creating the
+             * {@link DiskEncryptionSet} instance.
+             * @return the next stage of the definition
+             */
+            WithCreate withRBACBasedAccessToCurrentKeyVault();
```

Comment:
Should it be `Rbac`?

Reply:
I saw a deprecated method use this pattern for method name:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java#L635

So for abbreviations in the middle, we capitalize the first alphabet and leave the rest lowercase?

Reply:
maybe just call it `withRoleBasedAccessToCurrentKeyVault`..

Reply:
Yes, seems good!